api/envoy/extensions/request_id/uuid/v3/uuid.proto

```diff
@@ -16,13 +16,23 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
 // Configuration for the default UUID request ID extension which has the following behavior:
 //
-// 1. Request ID is propagated using the *x-request-id* header.
+// 1. Request ID is propagated using the :ref:`x-request-id
+//    <config_http_conn_man_headers_x-request-id>` header.
 //
-// 2. Request ID is a universally unique identifier (UUID).
+// 2. Request ID is a universally unique identifier `(UUID4)
+//    <https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random)>`_.
 //
 // 3. Tracing decision (sampled, forced, etc) is set in 14th byte of the UUID. By default this will
 //    overwrite existing UUIDs received in the *x-request-id* header if the trace sampling decision
-//    is changed.
+//    is changed. The 14th byte of the UUID4 has been chosen because it is fixed to '4' by the
```

Comment:
Also this should really be s/byte/nibble I think. I can make this change in the next round of comments.